app/src/main/java/com/pvp/app/ui/screen/task/TaskScreen.kt

```diff
@@ -1,18 +1,222 @@
+            textAlign = TextAlign.Left,
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 8.dp)
+        )
+    }
+}
 @Composable
 fun TaskBox(
     task: Task
 ) {
 
+    // Later task.isCompleted should be used
+    var checked by remember {
+        mutableStateOf(false)
+    }
+
+    Card(
+        shape = RectangleShape,
+        modifier = Modifier
+            .fillMaxWidth()
+            .border(BorderStroke(1.dp, MaterialTheme.colorScheme.outline))
+    ) {
+
+        Column(
+        ) {
+
+            Row(
+                modifier = Modifier.fillMaxWidth(),
+            ) {
+                Checkbox(
+                    checked = checked,
+                    onCheckedChange = {checked = it},
+                    modifier = Modifier
+                        .size(36.dp)
+                        .align(CenterVertically)
+                )
+                Text(
+                    task.title,
+                    textAlign = TextAlign.Center,
+                    modifier = Modifier
+                        .align(CenterVertically)
+                        .weight(1f),
+                    fontSize = 20.sp
+                )
+            }
+
+            Spacer(modifier = Modifier.height(4.dp))
+
+            when(task){
+                is SportTask -> SportTaskBoxBody(task)
+                is MealTask -> MealTaskBoxBody(task)
+                else -> TaskBoxBody(task)
+            }
+
+            val timeString = "Scheduled at ${task.scheduledAt
+                .toLocalTime()
+                .format(DateTimeFormatter.ofPattern("HH:mm"))}"
+
+            Text(
+                timeString,
+                modifier = Modifier
+                    .padding(start = 8.dp)
+            )
+        }
+    }
 }
 
 @Composable
 fun TaskScreen(
     task: Task
 ) {
+    // Only for testing purposes, delete later
+    // To add preview, comment out task: Task from constructor.
+    val mealTask = MealTask(
+        description = "Prepare dinner",
+        duration = Duration.ofMinutes(48),
+        id = "1",
+        isCompleted = false,
+        recipe = "Chicken breast",
+        scheduledAt = LocalDateTime.now(),
+        title = "Cook Dinner",
+        userEmail = "example@example.com"
+    )
+
+    val sportTask = SportTask(
+        activity = SportActivity.Running,
+        description = "Run in the park",
+        distance = 5.0,
+        duration = Duration.ofMinutes(65),
+        id = "2",
+        isCompleted = false,
+        scheduledAt = LocalDateTime.now(),
+        title = "Morning Run",
+        userEmail = "example@example.com"
+    )
 
+    val task = Task(
+        description = "Complete project tasks",
+        duration = Duration.ofHours(2),
+        id = "3",
+        isCompleted = false,
+        scheduledAt = LocalDateTime.now(),
+        title = "Project Tasks",
+        userEmail = "example@example.com"
+    )
+    TaskBox(sportTask)
```

Comment:
Single empty line should be added to separate different elements (variable, component).
Also, a single empty line (line-break) could be added to the end of the file to support UNIX file endings correctly as everywhere else.